lib/application_controller_patch.rb

```diff
@@ -0,0 +1,28 @@
+module ApplicationControllerPatch
+    def self.included(base)
+        base.send(:include, InstanceMethods)
+        base.send(:after_action, :remove_lazy_loading)
+        base.send(:after_action, :collapsible_images)
+    end
+     
+    module InstanceMethods
+        def remove_lazy_loading
+            if RedmineGoodiesSettings.remove_lazy_loading?
+                response.body = response.body.gsub(/loading="lazy"/, "")
+            end
+        end
+
+        def collapsible_images
+            if RedmineGoodiesSettings.collapsible_images?
+                # e.g. <div class="..."><p><img src="/attachments/download/90/clipboard-202408191804-gdzhu.png" alt=""></p></div>, this regex will match: 
+                # "<p><img src="/attachments/download/90/clipboard-202408191804-gdzhu.png" alt=""></p>"
+                response.body = response.body.gsub(/<p><img\s+[^>]*src=["']?([^"'\s>]+)["']?[^>]*>/) do |img|
+                    "<details open><summary>Click to expand image | <a href=\"#\" onclick=\"expandAll('#main'); return false;\">Expand all</a> | <a href=\"#\" onclick=\"collapseAll('#main'); return false;\">Collapse all</a></summary>#{img}</details>"
+                end
+                response.body = response.body.gsub(/<\/body>/) do |body_end|
+                    "<script>getIsCollapsedValueFromLocalStorage()</script>#{body_end}"
```

Comment:
I understand that you want to call this function on every page generated by `ApplicationController`.

I think the better way would be to put it in the partial `_collapsible_images.html.erb`, at the end, behind an if (for setting), and put it to execute after the document has loaded (otherwise, it would have nothing to collapse).

Of course, if you don't like this solution, you could also add a partial for the footer (as you have done for the header), and add the call of the function in it.

Reply:
Done!